sql/src/main/java/io/crate/executor/transport/task/elasticsearch/ESClusterUpdateSettingsTask.java

```diff
@@ -23,42 +23,69 @@
 
 import com.google.common.base.Function;
 import com.google.common.base.Functions;
+import com.google.common.collect.Iterables;
 import io.crate.core.collections.Row;
 import io.crate.core.collections.Row1;
 import io.crate.executor.JobTask;
 import io.crate.executor.transport.OneRowActionListener;
+import io.crate.metadata.settings.CrateSettings;
+import io.crate.metadata.settings.SettingsApplier;
 import io.crate.operation.projectors.RowReceiver;
 import io.crate.planner.node.ddl.ESClusterUpdateSettingsPlan;
+import io.crate.sql.tree.Expression;
 import org.elasticsearch.action.admin.cluster.settings.ClusterUpdateSettingsRequest;
 import org.elasticsearch.action.admin.cluster.settings.ClusterUpdateSettingsResponse;
 import org.elasticsearch.action.admin.cluster.settings.TransportClusterUpdateSettingsAction;
+import org.elasticsearch.common.settings.Settings;
+
+import java.util.List;
+import java.util.Map;
 
 public class ESClusterUpdateSettingsTask extends JobTask {
 
     private static final Function<Object, Row> TO_ONE_ROW = Functions.<Row>constant(new Row1(1L));
-    ;
+
+    private final ESClusterUpdateSettingsPlan plan;
     private final TransportClusterUpdateSettingsAction transport;
-    private final ClusterUpdateSettingsRequest request;
 
     public ESClusterUpdateSettingsTask(ESClusterUpdateSettingsPlan plan,
                                        TransportClusterUpdateSettingsAction transport) {
         super(plan.jobId());
+        this.plan = plan;
         this.transport = transport;
+    }
+
+    @Override
+    public void execute(RowReceiver rowReceiver, Row parameters) {
+        ClusterUpdateSettingsRequest request = buildESUpdateClusterSettingRequest(
+            buildSettingsFrom(plan.persistentSettings(), parameters),
+            buildSettingsFrom(plan.transientSettings(), parameters)
+        );
+        OneRowActionListener<ClusterUpdateSettingsResponse> actionListener = new OneRowActionListener<>(rowReceiver, TO_ONE_ROW);
+        transport.execute(request, actionListener);
+    }
 
-        request = new ClusterUpdateSettingsRequest();
-        request.persistentSettings(plan.persistentSettings());
-        request.transientSettings(plan.transientSettings());
+    Settings buildSettingsFrom(Map<String, List<Expression>> settingsMap, Row parameters) {
```

Comment:
Can this be static? Then you don't need to create the task to unit-test it and avoid having to mock the dependencies